### ISSUE TITLE
Got problem when have blank lines in a task list

### ISSUE THREAD

**Issue by PaleNeutron** [FND/markdown-checklist#7]
In this test code, I add a blank line in the task list and the parser failed.

```
>>> text = '''- [ ] something
    - [ ] another

- [ ] one
    - [ ] another'''
>>> print(markdown.markdown(text, ['markdown_checklist.extension']))
<ul>
<li>
<p>[ ] something</p>
<ul class="checklist">
<li><input type="checkbox" disabled> another</li>
</ul>
</li>
<li>
<p>[ ] one</p>
<ul class="checklist">
<li><input type="checkbox" disabled> another</li>
</ul>
</li>
</ul>
>>> text = '''- [ ] something
    - [ ] another
- [ ] one
    - [ ] another'''
>>> print(markdown.markdown(text, ['markdown_checklist.extension']))
<ul class="checklist">
<li><input type="checkbox" disabled> something<ul class="checklist">
<li><input type="checkbox" disabled> another</li>
</ul>
</li>
<li><input type="checkbox" disabled> one<ul class="checklist">
<li><input type="checkbox" disabled> another</li>
</ul>
</li>
</ul>
```

**Comment by FND** [comment]
I'm afraid this isn't something I can fix easily, as those extra `p`s result from [Python-Markdown](https://github.com/waylan/Python-Markdown)'s parsing (which is in line with how the orginal Markdown implementation works, [CommonMark](http://spec.commonmark.org/dingus/) seems to agree).

I'm also not sure what the right behavior here should be - perhaps @cdent has an opinion?

**Comment by PaleNeutron** [comment]
Thanks, I have read your code and found the problem isn't here, but I really do not want to dive into the hold project's code. Someone familiar with the project can fix it much more easier. Should I open an issue in `Python-MarkDown` project?

**Comment by cdent** [comment]
I would suggest narrowing the test case to leave out the checklist markup (that is, just make a simple list) and see what the behavior is then and compare with other implementations.

**Comment by FND** [comment]
AIUI this behavior (a blank line causing a `p` element to appear within `li`) is how Markdown was designed, for better or worse. So I suspect you won't have much luck complaining to library authors - might still be worth asking though, if only for personal edification.

That brings me back to my original(ly badly phrased) question: Should Markdown Checklist compensate for this quirk, treating `<li><p>[ ] ... </p></li>` as `<li><p>[ ] ... </p></li>`? (As noted, I'm not sure that's a realistic option either way.)

**Comment by radude** [comment]
Had the same problem, so I've made an extension: https://github.com/radude/mdx_truly_sane_lists

Test code:

```python
from markdown import markdown

text = '''- [ ] something
    - [ ] another

- [ ] one
    - [ ] another'''

print(markdown(text=text,
               extensions=[
                   'markdown_checklist.extension',
                   'mdx_truly_sane_lists'
               ],
               extension_configs={
                   'mdx_truly_sane_lists': {
                       'nested_indent': 4,
                       'truly_sane': True,
                   }},
               )
      )
```

Output:

```html
<ul class="checklist">
<li><input type="checkbox" disabled> something<ul class="checklist">
<li><input type="checkbox" disabled> another</li>
</ul>
</li>
</ul>
<ul class="checklist">
<li><input type="checkbox" disabled> one<ul class="checklist">
<li><input type="checkbox" disabled> another</li>
</ul>
</li>
</ul>
```

Consider the issue fixed now? :P

**Comment by FND** [comment]
Thanks @radude, I'm glad that exists and have added a note to the README.

It appears that this discussion has run its course, so I'll go ahead and close this issue - feel free to reopen if there's anything further to discuss.

**Comment by radude** [comment]
@FND Thanks, but you shouldn't have added mdx_truly_sane_lists as an alternative to your checklists. It does not provide checklist support, but it fixes those extra `p`s from Python-Markdown's parsing (hence fixes this issue).

So if anyone want to have the same behavior that Neutron wants they will have to use mdx_truly_sane_lists and markdown-checklist **together**.

---

BTW, I believe my extension also fixes PR #11, but I didn't get much into this PR so would be nice if @ava-dylang could confirm it.

**Comment by FND** [comment]
Oh, right, I didn't quite pay attention there (it's been too long since I've used Python Markdown myself... ) - thanks for the heads-up.
